src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WinRTWebSocket.cs

```diff
@@ -105,6 +114,21 @@ public override string SubProtocol
                 websocketControl.SupportedProtocols.Add(subProtocol);
             }
 
+            if (options.ClientCertificates.Count > 0)
+            {
+                if (!MessageWebSocketClientCertificateSupported)
+                {
+                    throw new PlatformNotSupportedException(string.Format(
+                        CultureInfo.InvariantCulture,
+                        SR.net_WebSockets_UWPClientCertSupportRequiresWindows10InsiderPreviewBuild16215OrGreater));
+                }
+
+                RTCertificate winRtClientCert = ConvertDotNetClientCertToWinRtClientCert(options.ClientCertificates[0]);
```

Comment:
>Thanks for the context on why this is taking a collection of certs. It wasn't clear at all from the MSDN documentation.

The ClientWebSocketOptions object and ClientWebSocket itself came originally from .NET Framework.  And it was based on HttpWebRequest originally for the HTTP part and then a separate set of code to handle the WebSocket protocol.  HttpWebRequest HTTP stack takes a collection of client certs.  So, that is why the API uses a collection .... since it matches the convention of HttpWebRequest.